Fix link to product from order line item

cjuner opened the pull request:
### 1. Why is this change necessary?

The administration's client-side route "sw.product.detail" expects a supplied parameter to be a product id.

The product id is already properly used to generate the link in the context menu (instead of using the order line item "identifier").

While the previously used "identifier" from the order line item may match the product id, it may also be different.

Order line items created by the LineItemTransformer, given cart line items originally created by the CartLineItemController using the ProductLineItemFactory, will have such an "identifier" matching the product id.

However, that does not necessarily apply to "identifiers" of order line items created elsewhere. The OrderConverter from the SwagMigrationAssistant plugin is such a counterexample, as it sets the "identifier" to the order line item's id (not the product's id).



### 2. What does this change do, exactly?

This patch adjusts the link generation to use the known product id from
the order line item record.


### 3. Describe each step to reproduce the issue or behaviour.

1. Create an order with an order line item with an `identifier` that does not match its product id. Such an order line item can, for example, be created by using SwagMigrationAssistant to migrate a Shopware 5 shop to Shopware 6 (where the order line item's `id` will be used for the `identifier`), see: <https://github.com/shopware/SwagMigrationAssistant/blob/a32f19de2e1c5039d42250e64c8246bc90280d3f/Profile/Shopware/Converter/OrderConverter.php#L805>.
2. Open the orders overview in the administration. Open the order with the created order line item. Try to access the corresponding product by clicking the order line item's name in the order line items grid.
3. That clicked link on the order line item's name will lead to an empty products detail page in the administration and errors appear on the browser console, as there is no corresponding product with the id of the order line item's `identifier` value.


### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.**

aragon999 commented:
Hello, 

I am not sure how to write a proper test for this issue, sorry. The error I recieve is:
`( ! ) Error: Class 'Shopware\Bundle\AccountBundle\StoreFrontCustomerGreetingService' not found in /srv/http/sw530/var/cache/production____REVISION___/proxies/ShopwareProductionda39a3ee5e6b4b0d3255bfef95601890afd80709ProjectContainer.php on line 1460`
`( ! ) Fatal error: Uncaught Error: Class 'Shopware\Bundle\AccountBundle\StoreFrontCustomerGreetingService' not found in /srv/http/sw530/var/cache/production____REVISION___/proxies/ShopwareProductionda39a3ee5e6b4b0d3255bfef95601890afd80709ProjectContainer.php on line 1460`
You can find the generated proxy file here (zipped because Github does not allow `.php` files):
[ShopwareProductionda39a3ee5e6b4b0d3255bfef95601890afd80709ProjectContainer.zip](https://github.com/shopware/shopware/files/1138502/ShopwareProductionda39a3ee5e6b4b0d3255bfef95601890afd80709ProjectContainer.zip)

shyim commented:
I guess the composer optimized autoloader has fixed the issue. 
The file has the namespace Shopware\Bundle\AccountBundle; but the file is located in engine/Shopware/Bundle/AccountBundle/Service/StoreFrontCustomerGreetingService.php 👎

mitelg commented:
merged with https://github.com/shopware/shopware/commit/7ca877dd38490a1db018fec5b2ddf05019405e97

cjuner committed:
Fix link to product from order line item

The administration's client-side route "sw.product.detail" expects a
supplied parameter to be a product id.

This patch adjusts the link generation to use the known product id from
the order line item record.

The product id is already properly used to generate the link in the
context menu (instead of using the order line item "identifier").

While the previously used "identifier" from the order line item may
match the product id, it may also be different.

Order line items created by the LineItemTransformer, given cart line
items originally created by the CartLineItemController using the
ProductLineItemFactory, will have such an "identifier" matching the
product id.

However, that need not apply to "identifiers" of order line items
created elsewhere. The OrderConverter from the SwagMigrationAssistant
plugin is such a counterexample, as it sets the "identifier" to the
order line item's id (not the product's id).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=1193) <br/>All committers have signed the CLA.

mitelg commented:
thanks for your contribution @cjuner  :+1: :tada: :blue_heart:
